view/log_test.go

```diff
@@ -46,10 +84,76 @@ func TestLogIter(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	var i int
+	num, err := testNext(log.Now())
 
-	iter := log.Now()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if num != n*m {
+		t.Errorf("expected %d facts, got %d", n*m, num)
+	}
+}
+
+func TestMultiDomainLogIter(t *testing.T) {
+	domains := []string{"test", "another_test"}
+
+	// Transactions
+	n := 4
+
+	// Size of write
+	m := 5
 
+	engine := randMultidomainStorage(domains, n, m)
+
+	// Open and merge the commit logs.
+
+	log0, err := OpenLog(engine, domains[0], "commit")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	log1, err := OpenLog(engine, domains[1], "commit")
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	mergedStreams, err := Merge([]origins.Iterator{log0.Now(), log1.Now()})
```

Comment:
Not that it matters for a test, but it would be *safer* to do the following so the time is explicit across all logs being merged (unless of course the query is merging logs from different points in time).

```go
now := time.Now()
iter, err := Merge(log0.Asof(now), log1.Asof(now))
```